### PR TITLE
Fix Mono.awaitSingleOrNull non-compliance with 2.7

### PR DESCRIPTION
**Pull request by dkhalanskyjb**

### CODE REVIEW

File: reactive/kotlinx-coroutines-reactor/test/MonoAwaitCancellationStressTest.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.reactivestreams.*
+import reactor.core.*
+import reactor.core.publisher.*
+import java.util.concurrent.locks.*
+
+class MonoAwaitCancellationStressTest {
+    private val iterations = 10_000 * stressTestMultiplier
+
+    @Test
+    fun testAwaitCancellationOrder() = runBlocking {
```
Comment (qwwdfsad) [inline reactive/kotlinx-coroutines-reactor/test/MonoAwaitCancellationStressTest.kt:18]: This test consistently deadlocks on the very first iteration on my machine

Comment (dkhalanskyjb) [reply reactive/kotlinx-coroutines-reactor/test/MonoAwaitCancellationStressTest.kt:18]: For me, the behavior here is *extremely* odd. When I run the test, I get dozens of times in a row when *all* iterations pass, and then dozens of times in a row when, like for you, the first iteration hangs.

**Comment by OlegDokuka** [comment]
Just FYI. Having cancel racing with request is expected behavior but rule 2.7. Please see related PR I have done in the past for the spec see https://github.com/reactive-streams/reactive-streams-jvm/pull/489

**Comment by OlegDokuka** [comment]
@dkhalanskyjb rule 3.5 is not about race conditions but rather about idempotency and non-blocking behaviour

**Comment by OlegDokuka** [comment]
@qwwdfsad this PR is not really relevant, so my suggestion is to reject it.

**Comment by dkhalanskyjb** [comment]
Hey @OlegDokuka, any progress on updating the spec? I don't see any new proposals in https://github.com/reactive-streams/reactive-streams-jvm/pulls — is that the place to look?

**Comment by dkhalanskyjb** [comment]
Ok, I give up, this PR is certainly not useful anymore.

**Pull request closed without merging**